starter-code/app.js

```diff
@@ -1,5 +1,48 @@
+'use strict'
+
 const express = require('express');
 const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
 
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+// Retrieve a random chuck joke
+app.get('/random', (req, res, next) => {
+  client.getRandomJoke()
+    .then((joke) => {
+      let randomJ = {
+        pJoke: joke.value
+      }
+      res.render('index', randomJ);
+    }).catch((err) => {
+      console.log(error);
+      return next;
+    });
+});
+
+
+app.get('/categories', (req, res, next) => {
+  client.getJokeCategories()
+    .then((categories)=>  {
+      let data = {
+        categories
```

Comment:
It should work because of the ES6 notations but make sure you understand that the structure is something like this:
```
.then(( categoriesResponse )=>  {
    let data = { categories : categoriesResponse };
    res.render('index', data)
});
```